src/Core/Resolvers/Sql Query Structures/BaseSqlQueryStructure.cs

```diff
@@ -360,7 +360,7 @@ protected static object ParseParamAsSystemType(string param, Type systemType)
                 "Double" => double.Parse(param),
                 "Decimal" => decimal.Parse(param),
                 "Boolean" => bool.Parse(param),
-                "DateTime" => DateTimeOffset.Parse(param),
+                "DateTime" => DateTime.Parse(param),
```

Comment:
This is a bit of a tricky one. `DateTime` type is handled by HotChocolate as `DateTimeOffset`. Also, our db column type detection method gives `DateTime` for all `Date`, `DateTime`, and `DateTimeOffset`. Meaning that while the user can define non `DateTimeOffset` columns, they will be interpreted as `DateTimeOffset` and the gaps will be filled with 00:00:00 time and local offset.

Here is the Initial PR: https://github.com/Azure/data-api-builder/commit/63d5ef24e3bb90057fed015caa1d81f222de232a